### PR TITLE
[WIP]Chipstat

### PR DESCRIPTION
**Pull request by zz951**
mv ErrorPlots from ITSFhrTask.cxx to ITSFeeTask.cxx
add new Plots with ChipStatVsFeeId
mv ErrorPlots Check from ITSFhrCheck.cxx to ITSFeeCheck.cxx

### CODE REVIEW

File: Modules/ITS/src/ITSFeeTask.cxx [hunk 1]
```diff
@@ -408,6 +520,11 @@ void ITSFeeTask::monitorData(o2::framework::ProcessingContext& ctx)
     }
   }
 
+  for (int ierror = 0; ierror < o2::itsmft::GBTLinkDecodingStat::NErrorsDefined; ierror++) {
+    int feeError = mLinkErrorVsFeeid->Integral(1, mLinkErrorVsFeeid->GetXaxis()->GetNbins(), ierror + 1, ierror + 1);
+    mErrorPlots->SetBinContent(ierror + 1, feeError);
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFeeTask.cxx:525]: Is it expected that we have TH1 with the number of decoding errors only for  GBTLinkDecodingStat? @iravasen

**Comment by IsakovAD** [comment]
@iravasen Is it expected that we have TH1 with the number of the decoding errors only for GBTLinkDecodingStat but not for GBTChipDecodingStat?

File: Modules/ITS/src/ITSFeeTask.cxx (hunk 1 above)
Comment (iravasen) [inline Modules/ITS/src/ITSFeeTask.cxx:525]: can you please add the same plot which show the total number of errors for each error id also for `mChipErrorVsFeeid`?

**Comment by iravasen** [comment]
Hi @zz951, very nice piece of work! Thanks a lot! On top of the comments above, please remove from FhrCheck the check on the "ErrorPlots". And, when you will add the same plot for the other TH2 (on chipstat) please add the same check also for it in FeeCheck. Thanks.

**Comment by iravasen** [comment]
> @iravasen Is it expected that we have TH1 with the number of the decoding errors only for GBTLinkDecodingStat but not for GBTChipDecodingStat?

Hi @IsakovAD, thanks! No, I suggested in my comments above to add a plot also for GBTChipDecodingStat. The others are minor comments.

File: Modules/ITS/src/ITSFeeCheck.cxx [hunk 2]
```diff
@@ -138,6 +146,31 @@ void ITSFeeCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResul
 {
   TString status;
   int textColor;
+  TLatex* text[5];
+
+  if (mo->getName() == "General/ErrorPlots") {
+    auto* h = dynamic_cast<TH1D*>(mo->getObject());
+    if (checkResult == Quality::Good) {
+      text[0] = new TLatex(10, 0.6, "Quality::Good");
```
[the remark this answers is not in the document]
Comment (zz951) [reply Modules/ITS/src/ITSFeeCheck.cxx:154]: ok, i will modify it.

File: Modules/ITS/src/ITSFhrTask.cxx [hunk 3]
```diff
@@ -233,10 +234,18 @@ void ITSFhrTask::initialize(o2::framework::InitContext& /*ctx*/)
   }
 }
 
+/*void ITSFhrTask::createErrorTriggerPlots()
+{
+  mErrorPlots = new TH1D("General/ErrorPlots", "Decoding Errors", mNError, 0.5, mNError + 0.5);
+  mErrorPlots->SetMinimum(0);
+  mErrorPlots->SetFillColor(kRed);
+  getObjectsManager()->startPublishing(mErrorPlots); // mErrorPlots
+}*/
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFhrTask.cxx:243]: Please clear unnecessary comments for old mErrorPlots

Comment (IsakovAD) [reply Modules/ITS/src/ITSFhrTask.cxx:243]: The same for lines 348-350, 248, 528 etc

File: Modules/ITS/src/ITSFeeTask.cxx [hunk 4]
```diff
@@ -142,20 +142,25 @@ void ITSFeeTask::createFeePlots()
...
+  mChipErrorVsFeeid = new TH2I("General/ChipErrorVsFeeid", "Chip decoding errors per FeeId", (3 * StaveBoundary[3]) + (2 * (StaveBoundary[7] - StaveBoundary[3])), 0, (3 * StaveBoundary[3]) + (2 * (StaveBoundary[7] - StaveBoundary[3])), o2::itsmft::ChipStat::NErrorsDefined, 0.5, o2::itsmft::ChipStat::NErrorsDefined + 0.5);
   mChipErrorVsFeeid->SetMinimum(0);
   mChipErrorVsFeeid->SetStats(0);
   getObjectsManager()->startPublishing(mChipErrorVsFeeid);
 
-  mErrorPlots = new TH1D("General/ErrorPlots", "Decoding Errors", o2::itsmft::GBTLinkDecodingStat::NErrorsDefined, 0.5, o2::itsmft::GBTLinkDecodingStat::NErrorsDefined + 0.5);
-  mErrorPlots->SetMinimum(0);
-  mErrorPlots->SetFillColor(kRed);
-  getObjectsManager()->startPublishing(mErrorPlots); // mErrorPlots
+  mLinkErrorPlots = new TH1D("General/LinkErrorPlots", "GBTLink decodeing Errors", o2::itsmft::GBTLinkDecodingStat::NErrorsDefined, 0.5, o2::itsmft::GBTLinkDecodingStat::NErrorsDefined + 0.5);
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFeeTask.cxx:155]: Fix a typo "decodeing"

File: Modules/ITS/src/ITSFeeCheck.cxx [hunk 5]
```diff
@@ -146,32 +153,38 @@ void ITSFeeCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResul
...
-      text[0] = new TLatex(10, 0.7, "Quality::Bad");
-      text[1] = new TLatex(10, 0.5, "Decoding ERROR detected");
-      text[2] = new TLatex(10, 0.3, "Please Call Expert");
-      for (int i = 0; i < 3; ++i) {
-        text[i]->SetTextAlign(23);
-        text[i]->SetTextSize(0.08);
-        text[i]->SetTextColor(kRed);
-        h->GetListOfFunctions()->Add(text[i]);
-      }
+      status = "Quality::BAD (call expert)";
```
Comment (IsakovAD) [inline Modules/ITS/src/ITSFeeCheck.cxx:161]: Maybe it's good to set textcolor to kRed in case of BAD quality check

**Comment by iravasen** [comment]
Closing it for now since we discussed further items to be implemented (Zhen you will be informed by Artem soon).

**Pull request closed without merging**